### DISCUSSION TITLE
File upload link

### DISCUSSION THREAD

**Discussion by JCV67** [Budibase/budibase#13195] [Help]
Hello 
I am creating an app using postgres. Looks like file attach does not work with postgres even using bytea column type. Tried the Budibase-File-Upload plugin but I get a 'malformed array literal' error. 
Any way my goal is not to upload files but to store the file path (preferable the hyperlink ) of dragged windows file, is that possible?
Thank you

**Comment by ConorWebb96** [comment]
Hey @JCV67,

There are a couple of options to consider for managing file attachments:

1. **Internal Table with Common ID:** You can create an internal table within Budibase for uploading assets/files. These can be linked to your Postgres tables via a common ID stored in each table. By utilising two data providers, you can display the images related to the ID using a combination of both Budibase's internal table and your external Postgres table.
2. **External File Storage (e.g., S3)**: Alternatively, you can utilise an external file storage solution such as Amazon S3. Save the file names of the attachments inside your Postgres database. Then, retrieve the images by constructing the URL using the storage bucket URL and the file name stored in the database. Ensure that the bucket where the images are stored is publicly accessible, as if they aren't this method won't work and your attachments won't be displayed.

Both approaches offer their own advantages, so choose the one that best suits your requirements and infrastructure.

I hope either of these methods helps point you in the right direction.

**Reply by JCV67** [reply]
Thank you @ConorWebb96 . Tomorrow I'll try the first option.
